src/renderer/util/history.ts

```diff
@@ -0,0 +1,130 @@
+interface Operation {
+    execute: ()=>void;
+    undo: ()=>void;
+
+    next: Operation | null;
+    prev: Operation | null;
+}
+
+function createOperation(execute:()=>void, undo:()=>void): Operation {
+  return {
+    'execute': execute,
+    'undo': undo,
+    'next': null,
+    'prev': null
+  };
+}
+
+
+/**
+ * A stack for operations which you might want to undo/redo.
+ * 
+ * Has a max size, which once exceeded drops elements from the bottom of the stack.
+ */
+export class OperationUndoStack {
+  private top: Operation | null = null;
```

Comment:
TS `private` modifier was a temporary solution, see https://www.valentinog.com/blog/private/
"Modern" syntax is `#top`

Reply:
Ah I see...why cant they give me my C++ syntax :*(
Changed this to fit